examples/routing_service/routing_service_file_adapter/cpp/FileAdapter.cxx

```diff
@@ -0,0 +1,47 @@
+/*****************************************************************************/
+/*         (c) Copyright, Real-Time Innovations, All rights reserved.        */
+/*                                                                           */
+/*         Permission to modify and use for internal purposes granted.       */
+/* This software is provided "as is", without warranty, express or implied.  */
+/*                                                                           */
+/*****************************************************************************/
+
+#include "FileAdapter.hpp"
+#include "FileConnection.hpp"
+
+using namespace rti::community::examples;
+using namespace rti::routing;
+using namespace rti::routing::adapter;
+
+FileAdapter::FileAdapter(PropertySet &properties)
+{
+}
+
+Connection *FileAdapter::create_connection(
+        rti::routing::adapter::detail::StreamReaderListener
+                *input_stream_discovery_listener,
+        rti::routing::adapter::detail::StreamReaderListener
+                *output_stream_discovery_listener,
+        const PropertySet &properties)
+{
+    FileConnection *fc = new FileConnection(
+            input_stream_discovery_listener,
+            output_stream_discovery_listener,
+            properties);
+    return fc;
+};
+
+void FileAdapter::delete_connection(Connection *connection)
+{
+    /**
```

Comment:
you ned to call delete connection here, otherwise the connection object is leaking

Reply:
Done.